y24-055-todo.md

```diff
@@ -0,0 +1,82 @@
+
+  - called `dual_index_tag_set` with range `dual_index_tag_sets`
+  - called `dual_index_tag_well`
+
+- [x] Task 3b: 3a Tests `spec/data/sample_manifest_excel/columns.yml`
+
+- [x] Task 4a: Add new `dual_index_tag_sets` range in `config/sample_manifest_excel/ranges.yml`
+
+  - scope: :dual_index_tag_sets
+  - scope_on: TagSet
+
+- [x] Task 4b: 4a Tests `spec/data/sample_manifest_excel/ranges.yml`
+
+- [x] Task 5a: Tag Set model in `app/models/tag_set.rb`
+
+  - (Tag Sets link together two related tag groups, but it can also be used to represent single index tag sets)
+  - New method / scope called `dual_index_tag_sets` to filter only Dual Index Tag Sets
+  - AND visible?
+  - Option 1: Filter tag sets which have both a tag_group and tag2_group
+  - Option 2: ?
+
+- [x] Task 5b: 5a Tests `spec/models/tag_set_spec.rb`
+
+- [x] Task 6a: Specialised fields
+
+  - Create two new specialised fields files for `dual_index_tag_set` and `dual_index_tag_well`
+  - `dual_index_tag_set` - Check the Tag Set exists, and that it has at least 1 tag group
+  - `dual_index_tag_well`
+    - Check tag set has two tag groups
+    - check each tag group has a tag in the given well
+  - Tests
+
+- [x] Task 6b: Specialised fields
+
+  - on update, get the tag from each tag group well, and associate it with the sample in the plates well
+  - so that after manifest upload, the sample has two tags associated with it
+  - check against manufactor tag name specs with tags
+
+- [ ] Task 7: Update Column header text
+
+  - Rich to confirm the text for new manifest headers (dual_index_tag_set/ well)
+  - Update header and dialog, and column key if required
+  - Discussed renaming to "Tag Plate Name"
+
+- [ ] Task 8: New story - Well input validation
+
+  - Create a story
+  - to add validation of well position input in sample manifest upload,
+  - so that is it clear the input is wrong before adding all the data and uploading
+  - e.g only allow one letter (A-H) then 1 or 2 numbers (0-9)
+  - There is currently indirect validation, as the tag can't be found if there is an invalid well location
+  - But this could be could earlier if there is validation on the well location in the manifest
+
+- [ ] Task 9: New story - New adapter type
+
+  - Create a story
+  - To create a way to have Currently chromium manifest only show single index tag groups
+  - Current Chromium manifest shows dual index tag groups
+  - Possibly, create new adapter type "dual_index_tag_plate",
+  - and rename the existing to "single_index_tag_plate"
+  - Against each Tag Group in the DB, check if it is dual and should update to new adapter type
+  - Update Chromium tag group to be single_index_tag_plate adapter_type
+  - Therefore, Chromium adapter type should then only include single index tag plates
+
+- [ ] Questions
+
+- Does it matter which way round i5 and i7 for tag and tag2? Currently i7 is tag and i5 is tag2 after manifest upload
```

Comment:
It does. i7 is for the first. i5 is the second and forward index. I don't know the details but 7 and 5 refer to the positions on the sequencing adapter. i7 is also referred to as index 1 and i5 is referred to as index 2. We call them tag_group and tag2_group. So, tag and tag2 are correct after manifest upload respectively.